shared/ortcustomops.cc

```diff
@@ -1,11 +1,32 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include <mutex>
 #include <set>
 
 #include "onnxruntime_extensions.h"
 #include "ocos.h"
 
+struct OrtCustomOpDomainDeleter {
+  explicit OrtCustomOpDomainDeleter(const OrtApi* ort_api) {
+    ort_api_ = ort_api;
+  }
+  void operator()(OrtCustomOpDomain* domain) const {
+    ort_api_->ReleaseCustomOpDomain(domain);
+  }
+
+  const OrtApi* ort_api_;
+};
+
+using OrtCustomOpDomainUniquePtr = std::unique_ptr<OrtCustomOpDomain, OrtCustomOpDomainDeleter>;
+static std::vector<OrtCustomOpDomainUniquePtr> ort_custom_op_domain_container;
+static std::mutex ort_custom_op_domain_mutex;
+
+static void AddOrtCustomOpDomainToContainer(OrtCustomOpDomain* domain, const OrtApi* ort_api) {
+  std::lock_guard<std::mutex> lock(ort_custom_op_domain_mutex);
```

Comment:
Why does it need be thread-safe?

Reply:
In which case this function might be invoked concurrently from multiple threads?

Reply:
For example, a process with multiple onnxruntime sessions running in different threads. Each session might create its own OrtSessionOptions and register the onnxruntime-extensions custom ops

Reply:
Then how many times this function will be called?

Reply:
It will be called each time the user calls RegisterCustomOpsLibrary() with "ortcustomops.dll" as argument. And this can happen concurrently from different threads in a process (although I admit it's not a common scenario)